cppcache/test/QueueConnectionRequestTest.cpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <QueueConnectionRequest.hpp>
+
+#include <boost/process/environment.hpp>
+
+#include <gtest/gtest.h>
+
+#include <geode/DataOutput.hpp>
+
+#include "ByteArrayFixture.hpp"
+#include "DataOutputInternal.hpp"
+
+namespace apache {
+namespace geode {
+namespace client {
+
+inline std::string to_hex(const uint8_t* bytes, size_t len) {
+  std::stringstream ss;
+  ss << std::setfill('0') << std::hex;
+  for (size_t i(0); i < len; ++i) {
+    ss << std::setw(2) << static_cast<int>(bytes[i]);
+  }
+  return ss.str();
+}
+
+inline std::string to_hex(const DataOutput& out) {
+  return to_hex(out.getBuffer(), out.getBufferLength());
+}
+
+inline std::string int_to_hex_string(const int value) {
+  char hex[10];
+  sprintf(hex, "%x", value);
+  return std::string(hex);
+}
+
+TEST(QueueConnectionRequestTest, testToData) {
```

Comment:
While I love to see another unit test, it seems like there should have been a failing integration test. Some of the tests are disabled because they never worked. Perhaps there is one that matches this failure.

Reply:
I didn't create integration test, because It seemed to me that non of the functionality is affected with this bug beside debug logging in locator. Native client always sends QueueConnectionRequest with flag findDurable=false, so membershipId is never used for anything functional in locator, it is only logged at debug level.

I have put more details related to this in mails sent towards dev@geode.apache.org, mail Subject: Unable to get behavior described in documentation when using durable native client

Reply:
Oh, that's right, the bug was just the server side logging null pointer.